ansible/templates/nginx.conf.j2

```diff
@@ -59,7 +59,7 @@ server {
 server {
   server_name {{domain}};
   listen 80;
-  return 301 https://{{fqdn}}$request_uri;
+  return 301 https://{{canonical_domain}}$request_uri;
```

Comment:
This works as is, but it *would* be possible to have a single port 80 catchall that doesn't even specify a server name and redirects anything that hits it to the canonical.